src/main/java/com/github/tomakehurst/wiremock/http/RequestMethod.java

```diff
@@ -46,19 +47,18 @@ public static RequestMethod fromString(String value) {
     return new RequestMethod(value);
   }
 
+  @JsonCreator
+  public static Set<RequestMethod> fromSet(Set<String> values) {
+    return values.stream().map(RequestMethod::fromString).collect(Collectors.toSet());
+  }
+
   @JsonValue
   public String value() {
     return name;
   }
 
-  public boolean isOneOf(RequestMethod... methods) {
-    return asList(methods).contains(this);
-  }
-
   public MatchResult match(RequestMethod method) {
-    boolean getOrHeadMatch =
-        this.equals(GET_OR_HEAD) && (method.equals(GET) || method.equals(HEAD));
-    return MatchResult.of(this.equals(ANY) || this.equals(method) || getOrHeadMatch);
+    return MatchResult.of(this.equals(ANY) || this.equals(method));
```

Comment:
Looks like we've lost the correct GET_OR_HEAD matching here

Reply:
I modified the WireMock#getOrHead(UrlPattern urlPattern) method to
BasicMappingBuilder(Set.of(RequestMethod.GET, RequestMethod.HEAD), urlPattern)
The tests are working correctly.
WireMockClientAcceptanceTest#
testGetOrHeadRequestWhenGetMatchesShouldReturnAResponseBody
testGetOrHeadRequestWhenHeadMatchesShouldNotReturnAResponseBody
testGetOrHeadRequestWhenNoMethodNotMatchesShouldReturn404
testGetOrHeadRequestWhenPathDoesNotMatchShouldReturn404
I think that's enough.
What do you say?